ambry-frontend/src/test/java/com.github.ambry.frontend/PostBlobHandlerTest.java

```diff
@@ -345,4 +412,95 @@ private void doChunkUploadTest(int contentLength, boolean chunkUpload, String up
     return executionException -> assertEquals("Unexpected error code", errorCode,
         ((RouterException) executionException.getCause()).getErrorCode());
   }
+
+  // stitchBlobTest() helpers
+
+  /**
+   * Upload chunks using the router directly.
+   * @param creationTimeMs the creation time to set for the chunks.
+   * @param container the {@link Container} to create the chunks in.
+   * @param chunkSizes the sizes for each chunk to upload.
+   * @return a list of {@link ChunkInfo} objects that contains metadata about each chunk uploaded.
+   */
+  private List<ChunkInfo> uploadChunksViaRouter(long creationTimeMs, Container container, int... chunkSizes)
+      throws Exception {
+    long blobTtlSecs = TimeUnit.DAYS.toSeconds(1);
+    List<ChunkInfo> chunks = new ArrayList<>();
+    for (int chunkSize : chunkSizes) {
+      byte[] content = TestUtils.getRandomBytes(chunkSize);
+      BlobProperties blobProperties =
+          new BlobProperties(-1, SERVICE_ID, OWNER_ID, CONTENT_TYPE, !container.isCacheable(), blobTtlSecs,
+              creationTimeMs, container.getParentAccountId(), container.getId(), container.isEncrypted());
+      String blobId =
+          router.putBlob(blobProperties, null, new ByteBufferReadableStreamChannel(ByteBuffer.wrap(content)),
+              new PutBlobOptionsBuilder().chunkUpload(true).build()).get(TIMEOUT_SECS, TimeUnit.SECONDS);
+
+      chunks.add(new ChunkInfo(blobId, chunkSize, Utils.addSecondsToEpochTime(creationTimeMs, blobTtlSecs)));
+    }
+    return chunks;
+  }
+
+  /**
+   * Generate a signed ID for a data chunk based on the provided arguments.
+   * @param chunkInfo the {@link ChunkInfo} containing the blob ID and additional metadata.
+   * @param uploadSession the session ID to include in the metadata.
+   * @return the signed ID.
+   */
+  private String getSignedId(ChunkInfo chunkInfo, String uploadSession) {
+    Map<String, String> metadata = new HashMap<>(3);
+    metadata.put(RestUtils.Headers.BLOB_SIZE, Long.toString(chunkInfo.getChunkSizeInBytes()));
+    metadata.put(RestUtils.Headers.SESSION, uploadSession);
+    metadata.put(PostBlobHandler.EXPIRATION_TIME_MS_KEY, Long.toString(chunkInfo.getExpirationTimeInMs()));
+    try {
+      return "/" + idSigningService.getSignedId(chunkInfo.getBlobId(), metadata);
+    } catch (RestServiceException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * @param signedChunkIds the list of signed chunk IDs to include in the JSON body.
+   * @return a valid JSON body for a stitch request, encoded as UTF-8.
+   */
+  private byte[] getStitchRequestBody(List<String> signedChunkIds) {
+    return new JSONObject().put(PostBlobHandler.SIGNED_CHUNK_IDS_KEY, new JSONArray(signedChunkIds))
```

Comment:
I think it will useful to describe the json body format in `PostBlobHandler` at an appropriate place